tools/static-assets/skel/package.json

```diff
@@ -2,10 +2,20 @@
   "name": "~name~",
   "private": true,
   "scripts": {
-    "start": "meteor run"
+    "start": "meteor run",
+    "test": "meteor test --driver-package dispatch:mocha-browser",
+    "test-app": "meteor test --full-app --driver-package dispatch:mocha-browser",
+    "visualize": "meteor --production --extra-packages bundle-visualizer"
   },
   "dependencies": {
-    "@babel/runtime": "^7.0.0-beta.36",
+    "@babel/runtime": "^7.0.0-beta.40",
     "meteor-node-stubs": "^0.3.2"
+  },
+  "meteor": {
+    "mainModule": {
+      "client": "client/main.js",
+      "server": "server/main.js"
+    },
+    "testModule": "test/main.js"
```

Comment:
Since the intent of the `meteor create` boilerplate is to serve as a jump start for new applications, I'm wondering if it might make sense to restructure things to create initially separated `client` / `server` test starting points. So something like:

```
  "meteor": {
    "mainModule": {
      "client": "client/main.js",
      "server": "server/main.js"
    },
    "testModule": {
      "client": "client/tests.js",
      "server": "server/tests.js"
    }
  }
```

The create app is already pushing people towards using the `client` / `server` separated approach (instead of lumping things together with `Meteor.isClient` / `Meteor.isServer`), so we might want to do the same with testing.

Reply:
This is a reasonable point, though a `test` (or `tests`) root directory is a more common location in other app structures so perhaps `tests/client/main.js` and `tests/server/main.js` (and if someone wanted to put test helpers or common tests, they might gravitate toward putting them in the higher-level `tests/`.

Reply:
Tangentially, after the introduction of `mainModule`, and prior to the `testModule` config in `package.json:meteor`, I was tempted to open the discussion about a `meteor.usePackageJsonEntryPoints` (or similarly themed) boolean config that simply used the `package.json` `main` and `browser` attributes, rather than necessitate the `mainModule` definitions, with the intention being to align with other parts of the ecosystem, and minimize the changes necessary for those making a switch (to or from).

It's worth reminding myself that `mainModule` offers more flexibility since it supports many architectures ("client", "server", "web", "web.browser", "web.cordova", "os"), but I can't help but think a user who had just `meteor create`d for the first time, might wonder why we didn't just use `main` and `browser`.

Ultimately, I'm not sure if a `usePackageJsonEntryPoints` would simplify this skeleton or not, but that certainly crossed my mind here again.

Reply:
If I may, I certainly agree with @abernix about the fact that using `mainModule` may appeared weird for users coming here for the first time and discovering entry points provided by NPM `package.json` configuration file are not used.
I consider myself used to Meteor now, and I also wondered why introducing `mainModule` instead of simply using `main` when I first saw the related PR.
Due to the the multitude of supported architectures, I think it makes sense to introduce the `meteor{mainModule, testModule}` configuration, because it would probably be referenced in the first line of the documentation/guide new users will surely read at the beginning of their journey with Meteor.

However, in order to allow advanced users to use meteor on specific environment with `main` entry point (like server only usage for instance), I would suggest allowing to configure Meteor to use it, as it should be a hand up to the ecosystem way of doing things for common usage of entry points. In the same idea, why not use simply `meteor{main,test}` to define these entry points with the same naming convention used by npm `package.json` ? 

 ```
 "meteor": {
    "useEntryPoints" : true, // true or omitted uses the main and test objects to configure entry points, false use root main
    "main": {
      "client": "client/main.js",
      "server": "server/main.js"
    },
    "test": {
      "client": "client/tests.js",
      "server": "server/tests.js"
    }
  }
```

Reply:
The `mainModule` option was intended to mirror `api.mainModule` for Meteor packages, and `testModule` was intended to match `mainModule`. I don't think shortening their names has much benefit, because it would  make them harder to distinguish from other fields called `main` or `test` that might be meaningful outside of Meteor.

The `main` field would be tempting to re-use if it had any flexibility regarding client/server entry points, but it doesn't, because it's always just a single string. And the `browser` field is not something Node recognizes; it's a hack that client-side-only bundlers like Webpack and Browserify happen to use. I've [written at length](https://github.com/meteor/meteor/issues/6890#issuecomment-339817703) about my feelings on the `browser` field. I remain very reluctant to encourage its use in any way.

Another reason against using `main` is that it would suggest the Meteor app is somehow importable as a package in Node, or that importing the `main` module is the way to start the app, which is not accurate. That's the reason the application `package.json` never had a `main` field in the first place, and I don't think it makes sense to start now. The `meteor` section is clearly metadata for Meteor apps to use while you're running the `meteor` command-line tool. I'm pretty reluctant to be adding any configuration options at all, to be completely honest, so I really don't want to invest other parts of `package.json` with Meteor-specific meaning. Let's keep our designs confined to the `meteor` section if possible.

If we started using `main`, it wouldn't be long before people wondered about [the `module` field](https://github.com/rollup/rollup/wiki/pkg.module). While support for `module` is a reasonable thing to ask about, Meteor does not have the problem that the `main`/`module` disctintion was invented to solve. For Meteor, the `mainModule` and `testModule` options point to *source code* rather than compiled code, so there is no meaningful distinction between the CJS (`main`) entry point and the ESM (`module`) entry point. Using `main` to point to source code would be an abuse of what `main` means in Node.

If you think about it, the closest thing Meteor has to a Node-style `main` module is the `.meteor/local/build/main.js` file that Meteor runs internally when starting an app. Obviously I don't want to put `"main": ".meteor/local/build/main.js"` in `package.json`, but using `main` to point to anything else would be misleading/wrong.

In short, I think we're avoiding a lot of pointless complexity by focusing on a few simple but flexible Meteor-specific options in the `meteor` section of `package.json`, and I honestly think encouraging the use of `main` or `browser` would only constrain us in the future, without solving any problems in the present.